File: threedollar-api-user/src/main/java/com/depromeet/threedollar/api/service/store/dto/request/MenuRequest.java
package com.depromeet.threedollar.api.service.store.dto.request;

import com.depromeet.threedollar.domain.user.domain.store.Menu;
import com.depromeet.threedollar.domain.user.domain.store.MenuCategoryType;
import com.depromeet.threedollar.domain.user.domain.store.Store;
import lombok.*;
import org.hibernate.validator.constraints.Length;

import javax.validation.constraints.NotNull;

@EqualsAndHashCode
@ToString
@Getter
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class MenuRequest {

    @Length(max = 50, message = "{menu.name.length}")
    @NotNull(message = "{menu.name.notNull}")
    private String name;

    @Length(max = 100, message = "{menu.price.length}")
    @NotNull(message = "{menu.price.notNull}")
    private String price;

    @NotNull(message = "{menu.category.notNull}")
    private MenuCategoryType category;

    public static MenuRequest of(String name, String price, MenuCategoryType category) {
        return new MenuRequest(name, price, category);
    }

    public Menu toEntity(Store store) {
        return Menu.of(store, name, price, category);
    }

}

File: threedollar-api-boss-service/src/main/kotlin/com/depromeet/threedollar/api/bossservice/service/store/dto/request/MenuRequest.kt
package com.depromeet.threedollar.api.bossservice.service.store.dto.request

import com.depromeet.threedollar.domain.mongo.domain.bossservice.store.BossStoreMenu
import org.hibernate.validator.constraints.URL
import javax.validation.constraints.PositiveOrZero
import javax.validation.constraints.Size

data class MenuRequest(
    @field:Size(max = 20, message = "{store.menu.name.size}")
    val name: String,

    @field:PositiveOrZero(message = "{store.menu.price.positiveOrZero}")
    val price: Int = -1,

    @field:URL(message = "{store.imageUrl.url}")
    val imageUrl: String?,
) {

    fun toMenu(): BossStoreMenu {
        return BossStoreMenu.of(
            name = name,
            price = price,
            imageUrl = imageUrl
        )
    }

}
